WooCommerce/WooCommerceTests/Tools/MockOrders.swift

```diff
@@ -26,7 +26,8 @@ final class MockOrders {
                      items: [],
                      billingAddress: sampleAddress(),
                      shippingAddress: sampleAddress(),
-                     coupons: [])
+                     coupons: [],
+                     refunds: nil)
```

Comment:
That's an interesting thought. I would prime semantics over consistency here, I think it is more important that the model objects reflect the domain.

That being said, I am not sure what models the domain better here, when it comes to model an order that has not been refunded: a nil refund or an existing array of refunds that happens to be empty. 

There is a third option though, which is trying to void nil just to avoid nil 😛 

- https://thoughtbot.com/blog/if-you-gaze-into-nil-nil-gazes-also-into-you
- https://blog.ragnarson.com/2015/05/06/problems-with-nil.html
- https://medium.com/stantmob/avoiding-nil-represent-special-cases-as-objects-ca24c2a6bcbc

And my favourite:
https://www.youtube.com/watch?v=29MAL8pJImQ

Notice how all of those links are discussions about nil in Ruby 🤷‍♂ 

So, I guess, my point would be that, in a case where there is no a clear semantic reason to either use nil or have an empty array, maybe there are other considerations to avoid nil?

Reply:
@jaclync good point. I should have added the problem in the PR summary. I struggled with that one. It was easy with `coupons` and `items` because the response always returns the field. If there are no `coupons` or `items`, the field value always contains an array. But the refunds field may or may not exist at all.  _How do I represent fields that may not even exist in the response?_ I agree, an empty array sounds like a better representation.

> in a case where there is no a clear semantic reason to either use nil or have an empty array, maybe there are other considerations to avoid nil?

@ctarda true. Swift by design attempts to avoid nil, so I see your point.